app/utils/agendaitem-utils.js

```diff
@@ -120,10 +126,19 @@ export const setAgendaitemsNumber = async(agendaitems, isEditor, isDesignAgenda)
 export const reorderAgendaitemsOnAgenda = async(agenda, isEditor) => {
   await agenda.hasMany('agendaitems').reload();
   const agendaitems = await agenda.get('agendaitems');
-  const actualAgendaitems = agendaitems.filter((agendaitem) => !agendaitem.showAsRemark && !agendaitem.isDeleted)
-    .sortBy('number');
-  const actualAnnouncements = agendaitems.filter((agendaitem) => agendaitem.showAsRemark && !agendaitem.isDeleted)
-    .sortBy('number');
+  let actualAgendaitems = [];
+  let actualAnnouncements = [];
+  for (const agendaitem of agendaitems.sortBy('number').toArray()) {
+    if (agendaitem.isDeleted) {
+      continue;
```

Comment:
I personally prefer  `if (!agendaitem.isDeleted) { ... }` instead of an early return because that reads more _natural_